### ISSUE TITLE
append does not chain correctly

### ISSUE THREAD

**Issue by dpwrussell** [Olical/react-faux-dom#43]
Basic D3 example (not using `react-faux-dom`)
```js
    // #test is: <svg id="test"></svg>
    const svg = d3.select("#test")
      .attr({width, height})
      .append("circle")
      .attr("r", 4.5)
      .attr("cx", 60)
      .attr("cy", 60)
      .style("fill", "steelblue");
```
This works of course. This however, does not:
```js
    const svg = d3.select(ReactFauxDOM.createElement('svg'))
      .attr({width, height})
      .append("circle")
      .attr("r", 4.5)
      .attr("cx", 60)
      .attr("cy", 60)
      .style("fill", "steelblue");
    return svg.node().toReact();
```
I think the reason for this is that append does not correctly chain. What should happen is that the `append` acts like a selector and the following `attr` operates on the `circle`, but that is not what happens, they operate on the `svg`.

This works fine:
```js
    const svg = d3.select(ReactFauxDOM.createElement('svg'))
      .attr({width, height});
    svg.append("circle")
      .attr("r", 4.5)
      .attr("cx", 60)
      .attr("cy", 60)
      .style("fill", "steelblue");
    return svg.node().toReact();
```

I am not experienced with D3, but it would seem to me that usage of react-faux-dom is the only difference between the examples so I guess the problem lies there.

Cheers,

Douglas

**Comment by dpwrussell** [comment]
Chained `append`:
```html
<div id="mount">
  <circle style="fill:steelblue;" r="4.5" cx="60" cy="60" data-reactid=".0"></circle>
</div>
```

Non-chained append:
```html
<div id="mount">
  <svg width="2544" height="800" data-reactid=".0">
    <circle style="fill:steelblue;" r="4.5" cx="60" cy="60" data-reactid=".0.$faux-dom-0"></circle>
  </svg>
</div>
```

**Comment by Olical** [comment]
Thanks for the report! I'll get it tested and fixed ASAP. I'm just away
right now.

I'd put it down to one of my DOM methods returning the wrong thing, like
appending a child needs to return the child, not the parent you appended it
to.

**Comment by dpwrussell** [comment]
I've just realised what has happened, which is not a bug. In fact it is precisely because it is chaining correctly that there was a problem.

With and without react-faux-dom I assigned the result to the `svg` variable, but in the case where I did not use react-faux-dom, I never used it for anything so I didn't notice that `svg` has been assigned to `circle`, not the result of the select. I guess this is a lesson that a bit more thought than assigning existing selects to a variable and expecting them to work is required! My D3 inexperience at fault here.
